WordPress/src/androidTest/java/org/wordpress/android/networking/MenusRestWPComTest.java

```diff
@@ -0,0 +1,407 @@
+                Handler mainLooperHandler = new Handler(mTargetContext.getMainLooper());
+                mainLooperHandler.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        mTestRest = new MenusRestWPCom(new IMenusDelegate() {
+                            @Override public long getSiteId() {
+                                return Long.valueOf(WordPress.getCurrentRemoteBlogId());
+                            }
+                            @Override public void onMenuDeleted(int statusCode, MenuModel menu, boolean deleted) {
+                                success[0] = !deleted;
+                                countDown();
+                            }
+                            @Override public Context getContext() { return mTargetContext; }
+                            @Override public void onMenuUpdated(int statusCode, MenuModel menu) { countDown(); }
+                            @Override public void onMenuCreated(int statusCode, MenuModel menu) { countDown(); }
+                            @Override public void onMenusReceived(int statusCode, List<MenuModel> menus) { countDown(); }
+                            @Override public void onMenuReceived(int statusCode, MenuModel menu) { countDown(); }
+                        });
+                        Assert.assertTrue(mTestRest.deleteMenu(badMenu));
+                    }
+                });
+            }
+        };
+        mLatch.await(10, TimeUnit.SECONDS);
+        Assert.assertTrue(success[0]);
+    }
+
+    public void testDeleteNullMenu() {
+        mTestRest = new MenusRestWPCom(EMPTY_DELEGATE);
+        Assert.assertFalse(mTestRest.deleteMenu(null));
+    }
+
+    @SuppressWarnings("unused")
+    public void onEventMainThread(CoreEvents.BlogListChanged event) {
+        if (mBlogListHandler != null) mBlogListHandler.eventRecieved(event);
+    }
+
+    private MenuModel getMenu(long id, String name) {
+        MenuModel menu = new MenuModel();
+        menu.menuId = id;
+        menu.name = name;
+        return menu;
+    }
+
+    private Callback loginAndUpdateBlogs = new Callback() {
+        @Override
+        public void onSuccess() {
+            GenericUpdateBlogListTask getBlogs = new GenericUpdateBlogListTask(mTargetContext);
+            getBlogs.execute();
+        }
+
+        @Override
+        public void onError(int errorMessageId, boolean twoStepCodeRequired, boolean httpAuthRequired, boolean erroneousSslCertificate) {
+            countDown();
+            Assert.fail();
+        }
+    };
+
+    private interface OnBlogListChanged {
+        void eventRecieved(CoreEvents.BlogListChanged event);
+    }
+    private OnBlogListChanged mBlogListHandler;
+
+    private CountDownLatch mLatch;
+    private void countDown() {
+        if (mLatch != null) mLatch.countDown();
+    }
+
+    private final IMenusDelegate EMPTY_DELEGATE = new IMenusDelegate() {
```

Comment:
Agreed, it's not even strictly following a delegation pattern! I must have spent too much time looking at the iOS menus branch 😝 
I'll rename it to `MenusListener` 👍